parsl/executors/high_throughput/executor.py

```diff
@@ -331,7 +338,8 @@ def submit(self, func, *args, **kwargs):
         Returns:
               Future
         """
-        task_id = uuid.uuid4()
+        self._task_counter += 1
```

Comment:
Can `submit` be called from multiple threads? Because `+=` is maybe not threadsafe: https://stackoverflow.com/questions/1717393/is-the-operator-thread-safe-in-python

If this is true, fix critical for functionality

Reply:
We already get a lock in the DFK at the launch stage to avoid this.